epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModel.java

```diff
@@ -137,7 +183,54 @@ public void bind(T view, EpoxyModel<?> previouslyBoundModel) {
    *
    * @see EpoxyAdapter#onViewRecycled(EpoxyViewHolder)
    */
-  public void unbind(T view) {
+  public void unbind(@NonNull T view) {
+  }
+
+  /**
+   * Called when the model has changed but still represents the same item. Return true if you want
+   * to animate these changes manually in the viewholder instead of simple cross fading or whatever
+   * animation the {@link android.support.v7.widget.RecyclerView.ItemAnimator) does by default.
+   *
+   * @return true if you wish to animate changes in the model
+   */
+  public boolean canReuseViewHolder(@NonNull EpoxyModel<?> previouslyBoundModel) {
+    return false;
+  }
+
+  /**
+   * Called if you returned true in {@link #canAnimateChanges()}
+   * Animate your changes in here so that the viewHolder will reflect the model after animation is
+   * finished, but do not start the animation yourself. Basically like {@link #bind(T)} but with
+   * animations.
+   * Duration will not be changed by the ItemAnimator or any calls to
+   * {@link android.support.v7.widget.RecyclerView.ItemAnimator#setChangeDuration(long)}, but start
+   * delay may be overwritten.
+   * Note that any listeners you set may be removed or the animation canceled at any time.
+   *
+   * @return non-null animator to be started by the ItemAnimator when it sees fit
+   */
+  @Nullable
+  public EpoxyItemAnimation bindChangesAnimated(@NonNull T view,
+      @NonNull EpoxyModel<?> previouslyBoundModel) {
+    return null;
+  }
+
+  /**
+   * Override this function if you want to handle interrupted change animations yourself.
+   * This function will be called when an already running custom change animation is interrupted by
+   * a more recent model change. If you do not implement special behaviour, unfinished animations
+   * will be run in sequence.
+   *
+   * @param interruptedAnimation the interrupted running change animation you returned in the
+   *                             previous call to {@link #animateChanges(), or null if no animation
+   *                             was running already. Guaranteed to be of the same type, so casting
+   *                             is safe.
+   * @return non-null animator to be started by the ItemAnimator when it sees fit
+   */
+  @Nullable
+  public EpoxyItemAnimation bindChangesAnimated(@NonNull T view,
```

Comment:
It would be great to only have to implement one method, but it seems we would have to make some changes so `EpoxyItemAnimation bindChangesAnimated(view, previouslyBoundModel, interruptedAnimation)` is actually called with a null animation?